### ISSUE TITLE
Accessibility in Table Component

### ISSUE THREAD

**Issue by Ajudiya13** [bvaughn/react-virtualized#1604]
## Bug Report

Hard-coded role props in the Table component.

### What is the current behavior?

Currently, two div elements have role props "grid" and "gridcell" respectively in Table Component. This causes accessibility limitation to use Table Component with another role props.

### What is the expected behavior?

The role should be props driven so that the user can have their own role props to have better and customized accessibility experience.

**Comment by laughnan** [comment]
We are also having this issue that we need to configure the `Table` component's `role` value to `table` instead of `grid` due to limitations with screenreaders.

**Comment by laughnan** [comment]
Added unit tests and pending review from @wuweiweiwu or @bvaughn ==> https://github.com/bvaughn/react-virtualized/pull/1691
